### PR TITLE
[4.0] Fix sample data UI

### PR DESCRIPTION
**Pull request by laoneo**
Fixes the sampledate success part. @Bakual can you please test if it works an on J3?

### CODE REVIEW

File: media/mod_sampledata/js/sampledata-process.js [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 	var sampledataAjax = function(type, steps, step) {
 		if (step > steps) {
-			$('.sampledata-' + type + ' .row-title').append('<span class="icon-publish"> </span>');
+			$('.sampledata-' + type + ' .row-title').append('<span class="fa fa-check"> </span>');
```
Comment (Quy) [inline media/mod_sampledata/js/sampledata-process.js:13]: Please add `aria-hidden="true"` to `<span class="fa fa-check">`.

**Comment by dgrammatiko** [comment]
@laneo here is some vanilla js:

```js
/**
 * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
 * @license    GNU General Public License version 2 or later; see LICENSE.txt
 */
var Joomla = window.Joomla || {};

!(function (Joomla) {
	"use strict";

	var inProgress = false;
	
	Joomla.sampledataAjax = function(type, steps, step) {
		if (step > steps) {
			document.querySelector('.sampledata-' + type + ' .row-title').innerHTML += '<span class="fa fa-check"> </span>';
			document.querySelector('.sampledata-progress-' + type).classList.add('d-none');
			inProgress = false;
			return;
		}

		var stepLi = document.createElement('li');
		var stepLiP = document.createElement('li');
		var stepLiPImg = document.createElement('img');
		var progress = document.querySelector(".sampledata-progress-" + type + " progress");

		stepLi.classList.add('sampledata-steps-' + type + '-' + step);
		stepLiP.classList.add('loader-image');
		stepLiP.classList.add('text-center');
		stepLiPImg.src = window.modSampledataIconProgress;
		stepLiPImg.setAttribute('width', 30);
		stepLiPImg.setAttribute('height', 30);

		stepLiP.appendChild(stepLiPImg);
		stepLi.appendChild(stepLiP);

		document.querySelector("div.sampledata-progress-" + type + " ul").appendChild(stepLi);

		Joomla.request({
			url: window.modSampledataUrl,
			method   : 'POST',
			data: {
				type: type,
				plugin: 'SampledataApplyStep' + step,
				step: step
			},
			perform  : true,
			onSuccess: function (response, xhr) {
				var response = JSON.parse(response);

				stepLi.querySelector('.loader-image').remove();

				if (response.success && response.data && response.data.length > 0) {
					var success, value, resultClass, $msg;

					// Display all messages that we got
					for(var i = 0, l = response.data.length; i < l; i++) {
						value   = response.data[i];
						success = value.success;
						resultClass = success ? 'success' : 'error';

						var divEl = document.createElement('div');
						divEl.innerHTML = value.message;
						divEl.classList.add('alert');
						divEl.classList.add('alert-' + resultClass);

						stepLi.appendChild(divEl);
					}

					// Update progress
					progress.value = step/steps;

					// Move on next step
					if (success) {
						step++;
						Joomla.sampledataAjax(type, steps, step);
					}

				} else {
					// Display error alert
					Joomla.renderMessages({'error': [Joomla.JText._('MOD_SAMPLEDATA_INVALID_RESPONSE')]}, '.sampledata-steps-' + type + '-' + step);

					inProgress = false;
				}
			},
			onError  : function (xhr) {
				alert('Something went wrong! Please close and reopen the browser and try again!');
			}
		});
	};

	Joomla.sampledataApply = function(el) {
		var type = el.getAttribute('data-type');
		var steps = el.getAttribute('data-steps');

		// Check whether the work in progress or we alredy proccessed with current item
		if (inProgress) {
			return;
		}
		if (el.getAttribute('data-processed')) {
			alert(Joomla.JText._('MOD_SAMPLEDATA_ITEM_ALREADY_PROCESSED'));
			return;
		}

		// Make sure that use run this not by random clicking on the page links
		if (!confirm(Joomla.JText._('MOD_SAMPLEDATA_CONFIRM_START'))) {
			return false;
		}

		// Turn on the progress container
		document.querySelector('.sampledata-progress-' + type).classList.remove('d-none');
		el.setAttribute('data-processed', true);

		inProgress = true;
		Joomla.sampledataAjax(type, steps, 1);
		return false;
	};

})(Joomla);
```

**Comment by C-Lodder** [comment]
Just use the code from here: https://github.com/joomla/joomla-cms/pull/19269

Was rewritten in Vanilla and fixed the UI

**Comment by laoneo** [comment]
@C-Lodder can you not reopen yours and I will close mine? Makes more sense instead of duplicating your effort.

**Comment by C-Lodder** [comment]
@laoneo done. Will fix the conflicts later on today

**Comment by laoneo** [comment]
Closing in favor of #19269. Please test it that we can get it in.

**Pull request closed without merging**